### PR TITLE
 Add pattern field support for object group type "LOG"

### PR DESCRIPTION
**Pull request by pavel-elkin-klarna**

### CODE REVIEW

File: chaossearch/client/operation_create_object_group.go [hunk 1]
```diff
@@ -71,6 +71,11 @@ func marshalCreateObjectGroupRequest(req *CreateObjectGroupRequest) ([]byte, err
 		body["partitionBy"] = req.PartitionBy
 	}
 
+	if req.Format == "JSON" {
+		var format = body["format"].(map[string]interface{})
+		format["pattern"] = req.Pattern
```
Comment (benzaita) [inline chaossearch/client/operation_create_object_group.go:76]: Shouldn’t we set the pattern when the type is log?

Comment (pavel-elkin-klarna) [reply chaossearch/client/operation_create_object_group.go:76]: You're right, fixed

File: chaossearch/resource_object_group.go [hunk 2]
```diff
@@ -65,6 +65,13 @@ func resourceObjectGroup() *schema.Resource {
 				ForceNew:     true,
 				ValidateFunc: validation.StringIsValidRegExp,
 			},
+			"pattern": {
+				Type:         schema.TypeString,
+				Default:      ".*",
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"json", "log"}, true),
```
Comment (benzaita) [inline chaossearch/resource_object_group.go]: this makes sure the pattern is either `log` or `json`, right? Is that what we want?

**Review by benzaita — COMMENTED** [review]
Regarding testing this, I updated the readme with instructions: https://github.com/benzaita/terraform-provider-chaossearch

File: chaossearch/resource_object_group.go (hunk 2 above)
Comment (pavel-elkin-klarna) [reply chaossearch/resource_object_group.go]: Yes, it checks that the string is in array

Comment (benzaita) [reply chaossearch/resource_object_group.go]: but `pattern` should be a regex, not `json` or `log`. I'm confused. Can you clarify?

**Review by benzaita — APPROVED** [review]

**Comment by benzaita** [comment]
@pavel-elkin-klarna let me know when testing is complete so I can merge

**Comment by benzaita** [comment]
I have removed the regex validation from `master` since the `go-pcre` library breaks the build -- it is not compatible with Darwin/Windows. If you want this validation, please open a new PR with a different library.

**Pull request merged**